.devcontainer/devcontainer.json

```diff
@@ -1,54 +1,51 @@
+  "features": {
+    "ghcr.io/devcontainers/features/common-utils:1": {
+      "username": "none",
+      "upgradePackages": false
+    }
+  },
+  // Set *default* container specific settings.json values on container create.
+  "customizations": {
+    "vscode": {
+      "settings": {
         "python.defaultInterpreterPath": "/venv/bin/python"
-    },
-    "customizations": {
-        "vscode": {
-            // Add the IDs of extensions you want installed when the container is created.
-            "extensions": [
-                "ms-python.python",
-                "tamasfe.even-better-toml",
-                "redhat.vscode-yaml",
-                "ryanluker.vscode-coverage-gutters"
-            ]
-        }
-    },
-    // Make sure the files we are mapping into the container exist on the host
-    "initializeCommand": "bash -c 'for i in $HOME/.inputrc; do [ -f $i ] || touch $i; done'",
-    "runArgs": [
-        "--net=host",
-        "--security-opt=label=type:container_runtime_t"
-    ],
-    "mounts": [
-        "source=${localEnv:HOME}/.ssh,target=/root/.ssh,type=bind",
-        "source=${localEnv:HOME}/.inputrc,target=/root/.inputrc,type=bind",
-        // map in home directory - not strictly necessary but useful
-        "source=${localEnv:HOME},target=${localEnv:HOME},type=bind,consistency=cached"
-    ],
-    // make the workspace folder the same inside and outside of the container
-    "workspaceMount": "source=${localWorkspaceFolder},target=${localWorkspaceFolder},type=bind",
-    "workspaceFolder": "${localWorkspaceFolder}",
-    // After the container is created, install the python project in editable form
-    "postCreateCommand": "pip install -e .[dev]"
-}
\ No newline at end of file
+      },
+      // Add the IDs of extensions you want installed when the container is created.
+      "extensions": [
+        "ms-python.python",
+        "tamasfe.even-better-toml",
+        "redhat.vscode-yaml",
+        "ryanluker.vscode-coverage-gutters"
+      ]
+    }
+  },
+  // Make sure the files we are mapping into the container exist on the host
+  "initializeCommand": "bash -c 'for i in $HOME/.inputrc; do [ -f $i ] || touch $i; done'",
+  "runArgs": ["--net=host", "--security-opt=label=type:container_runtime_t"],
+  "mounts": [
+    "source=${localEnv:HOME}/.ssh,target=/root/.ssh,type=bind",
+    "source=${localEnv:HOME}/.inputrc,target=/root/.inputrc,type=bind",
+    // map in home directory - not strictly necessary but useful
+    "source=${localEnv:HOME},target=${localEnv:HOME},type=bind,consistency=cached"
+  ],
+  // make the workspace folder the same inside and outside of the container
+  "workspaceMount": "source=${localWorkspaceFolder},target=${localWorkspaceFolder},type=bind",
+  "workspaceFolder": "${localWorkspaceFolder}",
+  // After the container is created, install the python project in editable form
+  "postCreateCommand": "pip install -e .[dev]"
+}
```

Comment:
These changes are not related and should be their own PR if you think they are required, possibly into the DiamondLightSource/python-copier-template.

Reply:
not valid point, as already this pattern is here. 

> each small change should be own PR

what difference does it make, it's just paperwork

https://github.com/DiamondLightSource/python-copier-template/blob/main/.devcontainer/devcontainer.json

Reply:
I see the point definitely for different part of the codebase that do different features.


but this small dev-env change has been requested by vscode. the change is to put:
`1  "vscode": {
   2  "settings": {
     3   "python.defaultInterpreterPath": "/venv/bin/python"
    4},`
    line 3 into vscode and not the global settings. 
    
    it's just git that's complaining about many lines